rfcs/049-catalogue-api-aggregations-modelling.md

```diff
@@ -0,0 +1,176 @@
+
+```
+AggregationBucket[T] {
+  data: T
+  count: Int
+  type: String
+}
+```
+
+where `T` is the same as the value appearing in the Work model.
+
+For example, languages on a work appear as follows:
+
+```
+{
+  "languages": [
+    {
+      "id": "eng",
+      "label": "English",
+      "type": "Language"
+    },
+    {
+      "id": "fre",
+      "label": "French",
+      "type": "Language"
+    }
+  ],
+  ...
+}
+```
+
+and we see that `Language` type in the aggregation buckets:
+
+```
+"aggregations": {
+  "languages": {
+    "buckets": [
+      {
+        "data": {
+          "id": "eng",
+          "label": "English",
+          "type": "Language"
+        },
+        "count": 691840,
+        "type": "AggregationBucket"
+      },
+      {
+        "data": {
+          "id": "fre",
+          "label": "French",
+          "type": "Language"
+        },
+        "count": 67187,
+        "type": "AggregationBucket"
+      },
+...
+```
+
+This presents a clean, consistent interface to clients – a value looks the same whether it's in a work or in an aggregation.
+
+We don't to change this behaviour.
+
+
+
+## The current implementation, and the problems it poses
+
+The API uses [Elasticsearch aggregations][es_aggs] to aggregate over a field in internal model.
+Elasticsearch will return single string values, which the API then serialises into the display model.
+
+For example, our language aggregation starts as [an ES terms aggregation][terms_agg] over the `data.languages.id` field:
+
+```http
+GET /works-indexed-2022-04-28/_search
+{
+  "aggs": {
+    "languages": {
+      "terms": { "field": "data.languages.id" }
+    }
+  }
+}
+```
+
+The Elasticsearch API response can only use a single string in its buckets, for example:
+
+```
+{
+  "aggregations" : {
+    "languages" : {
+      "buckets" : [
+        {
+          "key" : "eng",
+          "doc_count" : 691840
+        },
+        {
+          "key" : "fre",
+          "doc_count" : 67187
+        },
+        ...
+```
+
+The API has to contain enough internal/display model logic to interpret these values -- to know that, say, `eng` means English and it's serialised as id/label/type.
+This is precisely the sort of model coupling we're trying to get away from.
+
+For more complex types, we've had to jump through hoops to shoehorn aggregations into this approach -- e.g. contributor values are stored like `person:Henry Wellcome` because we need both the type and the label to return them correctly.
+
+It would be nice if we could remove this coupling and simplify how aggregations work in the API.
+
+[es_aggs]: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html
+[terms_agg]: https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html
+
+
+
+## Proposed solution
+
+We add a new field `query.aggregatableValues` to the Elasticsearch model.
```

Comment:
"Elasticsearch model" is a bit unclear - how about "to the documents that we store in Elasticsearch"?